File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/order/impl/OrderServiceImpl.java
package io.billie.shipmentnotifier.service.order.impl;

import io.billie.shipmentnotifier.config.ShipmentNotifierConfig;
import io.billie.shipmentnotifier.exception.OrderNotFoundException;
import io.billie.shipmentnotifier.model.order.Order;
import io.billie.shipmentnotifier.model.order.dao.OrderRepository;
import io.billie.shipmentnotifier.service.order.OrderService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@Service
@RequiredArgsConstructor
public class OrderServiceImpl implements OrderService {

	private final OrderRepository orderRepository;

	@Transactional(readOnly = true)
	public Order findByTrackingCodeAndMerchantNumber(String orderTrackingCode, String merchantNumber) throws OrderNotFoundException {
		return orderRepository.findByTrackingCodeAndMerchantNumber(orderTrackingCode, merchantNumber)
				.orElseThrow(() -> new OrderNotFoundException("invalid tracking number"));
	}
}

File: src/main/kotlin/io/billie/orders/service/OrderServiceImpl.kt
package io.billie.orders.service

import io.billie.orders.data.OrderRepository
import io.billie.orders.dto.OrderRequest
import io.billie.orders.execption.OrderNotFoundException
import io.billie.orders.model.Order
import org.springframework.stereotype.Service
import java.util.*

@Service
class OrderServiceImpl (val orderRepository: OrderRepository) : OrderService {
    override fun findOrderByUid(orderUid: UUID): Order {
        return orderRepository.findOrderByUid(orderUid)
                .orElseThrow { OrderNotFoundException ("order with Uid: $orderUid was not found!") }
    }

    override fun findOrders(): List<Order> {
        return orderRepository.findOrders()
    }

    override fun createOrder(orderRequest: OrderRequest): UUID {
        return orderRepository.createOrder(orderRequest)
    }
}